cluster-operator/src/main/java/io/strimzi/operator/cluster/model/metrics/StrimziMetricsReporterModel.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.metrics;
+
+import io.strimzi.api.kafka.model.common.HasConfigurableMetrics;
+import io.strimzi.api.kafka.model.common.metrics.StrimziMetricsReporter;
+import io.strimzi.operator.common.model.InvalidResourceException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Model for the Strimzi Metrics Reporter Kafka plugin.
+ */
+public class StrimziMetricsReporterModel implements MetricsModel {
+    /**
+     * Fully qualified class name of the Strimzi Kafka Prometheus Metrics Reporter.
+     */
+    private final boolean isEnabled;
+    private final List<String> allowList;
+
+    /**
+     * Constructs the Metrics Model for managing configurable metrics to Strimzi.
+     *
+     * @param spec Custom resource section configuring metrics.
+     */
+    public StrimziMetricsReporterModel(HasConfigurableMetrics spec) {
+        this(spec, List.of(".*"));
+    }
+
+        /**
+         * Constructs the Metrics Model for managing configurable metrics to Strimzi.
+         *
+         * @param spec Custom resource section configuring metrics.
+         * @param defaultAllowList Default allow list to be used when no value is provided.
+         */
+    public StrimziMetricsReporterModel(HasConfigurableMetrics spec, List<String> defaultAllowList) {
```

Comment:
I could be wrong, but does defaultAllowList need to be passed to the constructor? Can it just part of this class?

Reply:
Every component model has it's own and specific default allow list, so it needs to be parametric.